app/src/main/java/de/danoeh/antennapod/fragment/SubscriptionFragment.java

```diff
@@ -292,51 +328,41 @@ private int getDefaultNumOfColumns() {
         return getResources().getInteger(R.integer.subscriptions_default_num_of_columns);
     }
 
-    @Override
-    public void onCreateContextMenu(ContextMenu menu, View v, ContextMenu.ContextMenuInfo menuInfo) {
-        super.onCreateContextMenu(menu, v, menuInfo);
-        if (menuInfo == null) {
-            return;
-        }
-        AdapterView.AdapterContextMenuInfo adapterInfo = (AdapterView.AdapterContextMenuInfo) menuInfo;
-        int position = adapterInfo.position;
-
-        NavDrawerData.DrawerItem selectedObject = (NavDrawerData.DrawerItem) subscriptionAdapter.getItem(position);
-
-        if (selectedObject.type == NavDrawerData.DrawerItem.Type.FEED) {
-            MenuInflater inflater = requireActivity().getMenuInflater();
-            inflater.inflate(R.menu.nav_feed_context, menu);
-            selectedFeed = ((NavDrawerData.FeedDrawerItem) selectedObject).feed;
-        }
-        menu.setHeaderTitle(selectedObject.getTitle());
-    }
-
     @Override
     public boolean onContextItemSelected(MenuItem item) {
-        if (selectedFeed == null) {
+        Feed feed = subscriptionAdapter.getSelectedFeed();
+        if (feed == null) {
             return false;
         }
-
-        Feed feed = selectedFeed;
-        selectedFeed = null;
-        final int itemId = item.getItemId();
-        if (itemId == R.id.remove_all_new_flags_item) {
+        int itemId = item.getItemId();
+        if (itemId ==  R.id.remove_all_new_flags_item) {
             displayConfirmationDialog(
                     R.string.remove_all_new_flags_label,
                     R.string.remove_all_new_flags_confirmation_msg,
                     () -> DBWriter.removeFeedNewFlag(feed.getId()));
             return true;
-        } else if (itemId == R.id.add_to_folder) {
-            TagSettingsDialog.newInstance(feed.getPreferences()).show(getChildFragmentManager(), TagSettingsDialog.TAG);
+        }
+        if (itemId ==  R.id.mark_all_read_item) {
+            displayConfirmationDialog(
+                    R.string.mark_all_read_label,
+                    R.string.mark_all_read_confirmation_msg,
+                    () -> DBWriter.markFeedRead(feed.getId()));
             return true;
-        } else if (itemId == R.id.rename_item) {
+        }
+        if (itemId ==  R.id.rename_item) {
             new RenameFeedDialog(getActivity(), feed).show();
             return true;
-        } else if (itemId == R.id.remove_item) {
+        }
```

Comment:
Why did you change this to individual if statements? Only one of them can be executed. Using else if makes this more clear. Also, the change here is pretty much unrelated to the PR.

Reply:
If I use a switch I static analysis will report an error suggesting to use if statements, which you probably know why.